fairseq2n/src/fairseq2n/data/data_pipeline.cc

```diff
@@ -283,6 +285,34 @@ data_pipeline::count(std::int64_t start, std::optional<std::string> key)
     return data_pipeline_builder{std::move(factory)};
 }
 
+data_pipeline_builder
+data_pipeline::concat(
+    std::vector<data_pipeline> pipelines)
+{
+    if (pipelines.empty())
+        throw_<std::invalid_argument>(
+            "`pipelines` does not contain any elements. Can not concatenate from empty set.");
+
+    bool is_broken = std::any_of(
+        pipelines.begin(), pipelines.end(), [](const data_pipeline &pipeline)
+        {
+            return pipeline.is_broken();
+        });
+
+    if (is_broken)
+        throw_<std::invalid_argument>(
+            "At least one of the specified data pipelines is broken and cannot be used in concat.");
```

Comment:
Nit: I would end this sentence with "cannot be concatenated."

Reply:
Fixed